src/frontend/frontend/errors.js

```diff
@@ -22,28 +22,83 @@ angular.module('frontend-app')
   // Error service.
   .service('error-service', ['$rootScope', '$timeout', '$sce',
     function ($rootScope, $timeout, $sce) {
+      
     var self = this;
+
+    // global variables
     self.errors = [];
+    self.logs = [];
 
-    self.types = {
-      "seashell" : "If this error persists, please email <a href='mailto:seashell@cs.uwaterloo.ca'>seashell@cs.uwaterloo.ca</a> the error message, and the following information for debugging purposes:",
-      "marmoset" : "Make sure you can still access Marmoset's web interface, and try again in a few minutes.",
-      "webserver" : "Make sure you can access other websites located on the student.cs.uwaterloo.ca subdomain and try again in a few minutes."
-    };
+    // how many entries to keep in self.logs
+    var logSize = 200;
 
-    self.getMessage = function(type) {
-      return $sce.trustAsHtml(type ? self.types[type] : self.types.seashell);
-    };
+    // bug report email format
+    self.reportTo = "seashell@cs.uwaterloo.ca";
+    self.reportSubject = "seashell@cs.uwaterloo.ca";
+    self.reportBody = "Tell us what you were doing when this error showed up. \n[paste the log here]";
```

Comment:
I'm not sure how to do it.
The reportBody is a parameter in GET url and the logs are too long for that.